### ISSUE TITLE
11.0 LinePoolMixin Error

### ISSUE THREAD

**Issue by Talyrius** [semlar/WorldFlightMap#15]
It appears the `ObjectPoolMixin` was removed with patch 11.0 which broke your `LinePoolMixin`.

https://github.com/Gethe/wow-ui-source/blob/5a34f68e60458f344a776c3132d4d7431fd248a7/Interface/AddOns/Blizzard_SharedXMLBase/Pools.lua#L102-L109

**Comment by greerga** [comment]
Replace the file with:
```
local function LinePoolFactory(linePool)
    return linePool.parent:CreateLine(nil, linePool.layer, linePool.textureTemplate, linePool.subLayer)
end

function CreateLinePool(parent, layer, subLayer, textureTemplate, resetterFunc)
    local linePool = CreateSecureObjectPool(LinePoolFactory)
    linePool.parent = parent
    linePool.layer = layer
    linePool.subLayer = subLayer
    linePool.textureTemplate = textureTemplate
    return linePool
end
```

Edit: Securing was just changing "CreateObjectPool" to "CreateSecureObjectPool" fortunately.

**Comment by Drudoo** [comment]
@greerga Could you explain where to replace this? I tried to fix it in WorldFlightMap/LinePool.lua but that didn't fix the issue.

**Comment by greerga** [comment]
> @greerga Could you explain where to replace this? I tried to fix it in WorldFlightMap/LinePool.lua but that didn't fix the issue.

Yes, that's the place.  I'm not sure there aren't other taint issues in the WorldFlightMap code but it mostly fixed things.
